### ISSUE TITLE
Accessibility: Can't select TreeView items using UIAutomation SelectionItem.Select

### ISSUE THREAD

**Issue by leind375** [dotnet/winforms#9822]
### .NET version

8.0 Preview 7.

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

.NET 6 was working

### Issue description

Using UIAutomation API to select a TreeView item fails - the root item gets selected but not the desired item.

### Steps to reproduce

Create a basic WinForms app and add a TreeView. Edit the Nodes collection and add a root and 5 children. Run the app.
Using Inspect.exe (included in Visual Studio), expand the app and the tree and the root node. Click on one of the nodes then in the menu select Action=>SelectionItem.Select. Note that the node does not get selected. This is a regression from .NET 6 (didn't try 7).

**Comment by Olina-Zhang** [comment]
Can repro it in .Net 7.0 and .Net 8.0 RC2 latest build. Not repro in .Net 6.0.

**Comment by Tanya-Solyanik** [comment]
This is an LTS regression.

**Comment by Tanya-Solyanik** [comment]
Looks like we should enable support for SelectionItemProvider.

**Comment by MandiMan** [comment]
Verified this issue from main branch with winforms repo, it was fixed: Now using Inspect.exe, click on one of the nodes then in the menu select Action=>SelectionItem.Select, the node can be selected.

https://github.com/dotnet/winforms/assets/108860782/76e472ad-5236-460e-934e-3d66a6ade4f9

**Comment by Amy-Li03** [comment]
Verified this on .NET 8.0.100-rtm.23476.4, issue was fixed.

https://github.com/dotnet/winforms/assets/62929087/89e43330-3099-4c22-a1e0-afd6f3d9dea5

**Comment by Philip-Wang01** [comment]
Verified with .NET SDK 8.0.100-rc.2.23477.19 test pass build, this issue was fixed. Screenshot is same as above.
